src/App.tsx

```diff
@@ -14,6 +138,33 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const filtredList = todoList.filter(todo => {
+    switch (typeFilter) {
+      case 'Active':
+        return !todo.completed;
+      case 'Completed':
+        return todo.completed;
+      default:
+        return todo;
+    }
+  });
+
+  const handlerCloseErrors = () => {
```

Comment:
You can optimize your code using useCallback, useMemo, React.memo
For example, now you render todos list every time when writing some letter in the input